platform/default/sqlite_cache.cpp

```diff
@@ -470,16 +473,27 @@ void SQLiteCache::Impl::refresh(const Resource& resource, Seconds expires) {
     }
 }
 
-std::shared_ptr<SQLiteCache> SharedSQLiteCache::get(const std::string &path) {
-    std::shared_ptr<SQLiteCache> temp = masterPtr.lock();
-    if (!temp) {
-        temp.reset(new SQLiteCache(path));
-        masterPtr = temp;
+std::shared_ptr<SQLiteCache> SQLiteCache::getShared(const std::string &path) {
+    static std::mutex sharedMutex;
```

Comment:
Good point, I tried to mitigate that by locking on the mutex, but clearly didn't think this through.